x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/HttpClient.java

```diff
@@ -88,24 +60,32 @@ private static CloseableHttpAsyncClient createAsyncClient(PoolingNHttpClientConn
     }
 
     // Default for testing
-    HttpClient(HttpSettings settings, CloseableHttpAsyncClient asyncClient, IdleConnectionEvictor evictor, ThreadPool threadPool) {
+    HttpClient(HttpSettings settings, CloseableHttpAsyncClient asyncClient, ThreadPool threadPool) {
         this.settings = settings;
         this.threadPool = threadPool;
         this.client = asyncClient;
-        this.connectionEvictor = evictor;
     }
 
     public void start() {
         if (status.compareAndSet(Status.CREATED, Status.STARTED)) {
             client.start();
-            connectionEvictor.start();
         }
     }
 
-    public void send(HttpUriRequest request, ActionListener<HttpResult> listener) throws IOException {
+    public void send(HttpUriRequest request, ActionListener<HttpResult> listener) {
         // The caller must call start() first before attempting to send a request
         assert status.get() == Status.STARTED;
 
+        threadPool.executor(HTTP_CLIENT_SENDER_THREAD_POOL_NAME).execute(() -> {
```

Comment:
Would it be better to build our own executor service instead? One advantage I can see for building our own is defining a limit to the queue. I couldn't find a way to do that in the `ScalingExecutorBuilder`. Another option would be to use a `FixedExecutor` but that'll force at least 1 thread to be waiting around for tasks I think.

Reply:
Callers of this method could implement a queue if necessary and the callee is better placed to make decisions about retries etc. The important thing is that `doPrivilegedSend` can block acquiring a connection so the work should be done by the executor.

Reply:
Good point, I'll change this in a follow up PR 👍